src/lib/types.hpp

```diff
@@ -48,6 +45,14 @@ STRONG_TYPEDEF(opossum::ColumnID::base_type, ColumnCount);
 STRONG_TYPEDEF(uint32_t, ValueID);  // Cannot be larger than ChunkOffset
 STRONG_TYPEDEF(uint32_t, NodeID);
 STRONG_TYPEDEF(uint32_t, CpuID);
+STRONG_TYPEDEF(uint32_t, WorkerID);
+STRONG_TYPEDEF(uint32_t, TaskID);
+STRONG_TYPEDEF(uint32_t, ChunkOffset);
+
+// When changing these to 64-bit types, reading and writing to them might not be atomic anymore.
```

Comment:
I don't know if we should keep this comment. Every ARM starting with ARMv8-1a should support the necessary 64bit atomics. The M1 is covered, it appears to use at least the ARM v8.4-A instruction set.
BUT even the most recent Raspberry Pi versions do "only" have a ARM v8.0 instruction set. So here we would actually not run into problems as the atomics cover proper accesses, but 64-bit values wouldn't be as efficient as 32-bit ones here.

Please note that my hardware understanding is quite limited.

Reply:
What do you think about a comment like:
```
When changing these to 64-bit types, please be aware that not all platforms that Hyrise runs on support atomic 64-bit  instructions. Any Intel and AMD CPU of the last ten years should work fine. But 64-bit atomics on ARM CPUs might lead to less efficient code when using `std::atomic` on instructions sets earlier than ARMv8-1A (e.g., Raspberry Pi v3 or Amazon's Graviton1). Apple's M1 and Graviton2 support the required LSE instructions.
More information can be found here: https://community.arm.com/arm-community-blogs/b/tools-software-ides-blog/posts/making-the-most-of-the-arm-architecture-in-gcc-10
```

Reply:
I would shorten the comment a bit by removing the part `(e.g.,
Raspberry Pi v3 or Amazon's Graviton1). Apple's M1 and Graviton2 support the required LSE instructions`.

Reply:
Hmm, took me quite some time to understand the different ARM specs and which CPU supports which instruction set. Why would you remove it?

Reply:
Only for reference:
From my current understanding, x86 CPUs support atomic 64-bit operations via compare and exchange (CMPXCHG) instructions (https://en.wikipedia.org/wiki/Compare-and-swap#Implementations).

>Hmm, took me quite some time to understand the different ARM specs and which CPU supports which instruction set. Why would you remove it?

I think the key information regarding ARM CPUs is that on instruction sets earlier than ARMv8.1-A, atomic operations might be less efficient.
The Raspberry Pi v3 CPU and Graviton1 are then "only" CPU examples with earlier instruction sets, whereas M1 and Graviton2 are CPU examples with later instruction sets.

- Graviton1: https://en.wikichip.org/wiki/annapurna_labs/alpine/al73400
- Raspberry Pi v3 CPU: https://de.wikipedia.org/wiki/Arm_Cortex-A#Arm_Cortex-A53
- M1: https://en.wikichip.org/wiki/apple/mx/m1
- Graviton2: https://en.wikipedia.org/wiki/ARM_Cortex-A76#Neoverse_N1

While the instruction set version is an important note, the CPUs are only examples that may not be relevant in a few years and the comment may need to be updated. If a Hyrise user/developer wants to check for a target ARM chip to see if there are any inefficiencies in atomic operations, they can look up the instruction set for that chip IMO.

Reply:
`ARMv8-1A` --> `ARMv8.1-A`

Reply:
Convinced.

What do you think about:
```
When changing these to 64-bit types, please be aware that not all platforms that Hyrise runs on support atomic
64-bit instructions. Any Intel and AMD CPU since 2010 should work fine. For 64-bit atomics on ARM
CPUs, the instruction set should be at least ARMv8.1-A. Earlier instruction sets also work, but might yield less efficient code.
More information can be found here:
https://community.arm.com/arm-community-blogs/b/tools-software-ides-blog/posts/making-the-most-of-the-arm-architecture-in-gcc-10
```

Reply:
Looks good to me